packages/framework-core/src/booster-graphql-dispatcher.ts

```diff
@@ -49,7 +53,12 @@ export class BoosterGraphQLDispatcher {
       case 'CONNECT':
         return this.config.provider.graphQL.handleResult(null, graphQLWebsocketSubprotocolHeaders)
       case 'MESSAGE':
-        return this.config.provider.graphQL.handleResult(await this.handleMessage(envelopeOrError))
+        // eslint-disable-next-line no-case-declarations
```

Comment:
Looking again at the code, I noticed that the `GraphQLResolverContext` object doesn't depend on anything but the envelope. I think that we could initialize it at this point to avoid the changes in return types. I think that something like this is possible:

```typescript
case 'MESSAGE':
  const envelope = this.envelopeOrDie(envelopeOrError) // If we want to access the envelope here, we'd need to handle the errors. We could add a new method to do this.
  const context: GraphQLResolverContext = {
    connectionID: envelope.connectionID,
    responseHeaders: { ...this.config.defaultResponseHeaders },
    requestID: envelope.requestID,
    user: envelope.currentUser,
    // operation, <= This object can be added as we go deeper in the call stack
    pubSub: new NoopReadModelPubSub(),
    storeSubscriptions: true,
    context: envelope.context,
  }
  const response = this.handleMessage(envelope)
  return this.config.provider.graphQL.handleResult(response, context.responseHeaders)
```